algorithms/a3c/main.py

```diff
@@ -40,80 +54,276 @@
+        # use given config file to start config and then allow config_dict to overwrite values
+        config_file_dir_path = os.path.abspath(os.path.join(__file__, '../../..', 'gym_ai2thor',
+                                                            'config_files'))
+        args.config_file_path = os.path.join(config_file_dir_path, args.config_file_name)
+        env = AI2ThorEnv(config_file=args.config_file_path, config_dict=args.config_dict,
+                         seed=args.seed)  # todo decide
+        args.resolution = (env.config['resolution'][0], env.config['resolution'][1])
 
+    # Create shared model
+    if env.task.task_has_language_instructions:
+        # environment will return natural language sentence as part of state so process it with
+        # Gated Attention (GA) variant of A3C
+        shared_model = A3C_LSTM_GA(env.observation_space.shape[0], env.action_space.n,
+                                   args.resolution, len(env.task.word_to_idx),
+                                   args.max_episode_length)
+    else:
+        shared_model = ActorCritic(env.observation_space.shape[0], env.action_space.n,
+                                   args.resolution)
+    shared_model.share_memory()
+
+    # Create optimizer
     if args.no_shared:
         optimizer = None
     else:
         optimizer = my_optim.SharedAdam(shared_model.parameters(), lr=args.lr)
         optimizer.share_memory()
 
-    processes = []
+    env.close()  # above env initialisation was only to find certain params needed for models
 
-    counter = mp.Value('i', 0)
-    lock = mp.Lock()
+    # Checkpoint creation/loading below
+    checkpoint_counter = False
+    if not os.path.exists(args.checkpoint_path):
+        print('Tensorboard created experiment folder: {} and checkpoint folder '
+              'made here: {}'.format(args.experiment_path, args.checkpoint_path))
+        os.makedirs(args.checkpoint_path)
+    else:
+        print('Checkpoints path already exists at path: {}'.format(args.checkpoint_path))
+        # look for checkpoints and find one with largest total_length
+        checkpoint_paths = glob.glob(os.path.join(args.checkpoint_path, 'checkpoint_total_length*'))
+        if checkpoint_paths:
+            # Take checkpoint path with most experience
+            # e.g. 2000 from checkpoint_total_length_2000.pth.tar
+            checkpoint_file_name_ints = [
+                int(x.split('/')[-1].split('.pth.tar')[0].split('_')[-1])
+                for x in checkpoint_paths]
+            idx_of_latest = checkpoint_file_name_ints.index(max(checkpoint_file_name_ints))
+            checkpoint_to_load = checkpoint_paths[idx_of_latest]
+            print('Attempting to load latest checkpoint: {}'.format(checkpoint_to_load))
+
+            # load checkpoint and unpack values
+            if os.path.isfile(checkpoint_to_load):
+                print("Successfully loaded checkpoint {}".format(checkpoint_to_load))
```

Comment:
Before this print you didn't check if the checkpoint has valid data or anything. I would remove it completely or make a try except to catch `KeyError` exceptions